src/hash_map.rs

```diff
@@ -6,46 +6,177 @@ use std::iter::FromIterator;
 use std::ops::{Deref, DerefMut, Index};
 use std::panic::UnwindSafe;
 
+use crate::RandomState;
+
 /// A [`HashMap`](std::collections::HashMap) using [`RandomState`](crate::RandomState) to hash the items.
 /// (Requires the `std` feature to be enabled.)
 #[derive(Clone)]
 pub struct AHashMap<K, V, S = crate::RandomState>(HashMap<K, V, S>);
 
-impl<K, V> From<HashMap<K,V,crate::RandomState>> for AHashMap<K, V> {
-    fn from(item: HashMap<K,V,crate::RandomState>) -> Self {
+impl<K, V> From<HashMap<K, V, crate::RandomState>> for AHashMap<K, V> {
+    fn from(item: HashMap<K, V, crate::RandomState>) -> Self {
         AHashMap(item)
     }
 }
 
-impl<K, V> Into<HashMap<K,V,crate::RandomState>> for AHashMap<K, V> {
-    fn into(self) -> HashMap<K,V,crate::RandomState> {
+impl<K, V> Into<HashMap<K, V, crate::RandomState>> for AHashMap<K, V> {
+    fn into(self) -> HashMap<K, V, crate::RandomState> {
         self.0
     }
 }
 
-impl<K, V, S> AHashMap<K, V, S>
-where
-    S: BuildHasher + Default,
-{
+impl<K, V> AHashMap<K, V, RandomState> {
     pub fn new() -> Self {
-        AHashMap(HashMap::with_hasher(S::default()))
+        AHashMap(HashMap::with_hasher(RandomState::default()))
     }
 
     pub fn with_capacity(capacity: usize) -> Self {
-        AHashMap(HashMap::with_capacity_and_hasher(capacity, S::default()))
+        AHashMap(HashMap::with_capacity_and_hasher(capacity, RandomState::default()))
+    }
+
+    pub fn with_hasher(hash_builder: RandomState) -> Self {
+        AHashMap(HashMap::with_hasher(hash_builder))
+    }
+
+    pub fn with_capacity_and_hasher(capacity: usize, hash_builder: RandomState) -> Self {
+        AHashMap(HashMap::with_capacity_and_hasher(capacity, hash_builder))
     }
 }
 
-impl<K, V, S> AHashMap<K, V, S>
+impl<K, V> AHashMap<K, V, RandomState>
```

Comment:
I think you should keep the `S` parameter here: it doesn't change coercion but it enables to use the map with any hasher.